scripts/extensions/ai-widget/src/ai-assistant.tsx

```diff
@@ -84,14 +86,24 @@ export class AiAssistantWidget extends React.PureComponent<IArticleSideWidgetCom
         this.inactiveTabState = {};
         this.getDefaultState = this.getDefaultState.bind(this);
         this.setSection = this.setSection.bind(this);
+
+        const computeInitialState = () => {
+            if (this.props.initialState.currentTab != null) {
```

Comment:
why is conditional needed? I'd expect it would work the same regardless of state structure

Reply:
It's needed because of how state is saved through places. Sometimes through "SIDE_WIDGET" the currentTab is only saved. And other times when saved through `widget-react.tsx` the full state is saved. We could make the "SIDE_WIDGET" save the whole state, or we can also make a custom parser for each widget, to process its state accordingly. You might quickly say option 1, for now I think it'd be best too, but for the future we should consider option 2 as well because of nesting state inside the widgets.

Reply:
> It's needed because of how state is saved through places. Sometimes through "SIDE_WIDGET" the currentTab is only saved.

Could you save the entire state there? It's best if we can use whatever state structure we need in each widget and it doesn't impact code outside of the widget. Would that work? In that case I don't think we need any custom parsers for anything. Or am I missing something?

Reply:
Would work totally, I already did that, custom parsers would be for future proofing I meant to say, for now of course we don't need them